src/Abstractions/NexusMods.Abstractions.Resources.Caching/InMemoryStore.cs

```diff
@@ -0,0 +1,243 @@
+                Data = data,
+            };
+        }
+
+        _dictionary.Add(tuple, resource.Data);
+        return resource;
+    }
+
+    private bool _isDisposed;
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        if (_isDisposed) return;
+        _isDisposed = true;
+
+        _cts?.Dispose();
+
+        _dictionarySemaphore.Dispose();
+
+        foreach (var kv in _dictionary)
+        {
+            var disposable = kv.Value as IDisposable;
+            disposable?.Dispose();
+        }
+
+        _dictionary.Clear();
+    }
+}
+
+public static partial class ExtensionsMethods
+{
+    /// <summary>
+    /// Store the results in-memory.
+    /// </summary>
+    public static IResourceLoader<TResourceIdentifier, TData> StoreInMemory<TResourceIdentifier, TKey, TData>(
+        this IResourceLoader<TResourceIdentifier, TData> inner,
+        Func<TResourceIdentifier, TKey> keySelector,
+        IEqualityComparer<TKey> keyComparer,
+        InMemoryStore<TResourceIdentifier, TResourceIdentifier, TKey, TData>.ShouldDeleteKey? shouldDeleteKey,
+        Optional<TimeSpan> deleteDelay = default)
+        where TResourceIdentifier : notnull
+        where TData : notnull
+        where TKey : notnull
+    {
+        return inner.Then(
+            state: (keySelector, keyComparer, shouldDeleteKey, deleteDelay),
+            factory: static (input, inner) => new InMemoryStore<TResourceIdentifier, TResourceIdentifier, TKey, TData>(
+                keySelector: input.keySelector,
+                resourceIdentifierSelector: static x => x,
+                keyComparer: input.keyComparer,
+                shouldDeleteKey: input.shouldDeleteKey,
+                deleteDelay: input.deleteDelay,
+                inner: inner
+            )
+        );
+    }
+
+    /// <summary>
+    /// Store the results in-memory.
+    /// </summary>
+    public static IResourceLoader<TResourceIdentifier, TData> StoreInMemory<TResourceIdentifier, TKey, TData>(
+        this IResourceLoader<TKey, TData> inner,
+        Func<TResourceIdentifier, TKey> selector,
+        IEqualityComparer<TKey> keyComparer,
+        InMemoryStore<TResourceIdentifier, TKey, TKey, TData>.ShouldDeleteKey? shouldDeleteKey,
+        Optional<TimeSpan> deleteDelay = default)
+        where TResourceIdentifier : notnull
+        where TData : notnull
+        where TKey : notnull
+    {
+        return inner.Then(
+            state: (selector, keyComparer, shouldDeleteKey, deleteDelay),
+            factory: static (input, inner) => new InMemoryStore<TResourceIdentifier, TKey, TKey, TData>(
+                keySelector: input.selector,
+                resourceIdentifierSelector: input.selector,
+                keyComparer: input.keyComparer,
+                shouldDeleteKey: input.shouldDeleteKey,
+                deleteDelay: input.deleteDelay,
+                inner: inner
+            )
+        );
+    }
+
+    /// <summary>
+    /// Store the results in-memory.
```

Comment:
The code is good, but we still need to explain the relation between `StoreInMemory` (`InMemoryStore`), `UseCache` (`ResourceCache`) and `UseScopedCache` (`ScopedResourceCache`). Both are technically memory stores. 

Maybe link to https://github.com/bitfaster/BitFaster.Caching/wiki/IDisposable-and-Scoped-values#scoped for the scoped case. i.e. 'Prevent issues with disposal when there are multiple concurrent owners'/